scapy/layers/sixlowpan.py

```diff
@@ -102,96 +125,255 @@ def getfield(self, pkt, s):
+# https://tools.ietf.org/html/rfc4944#section-10.1
+# This implementation is NOT RECOMMENDED according to RFC 6282
 
 
-class LoWPANUncompressedIPv6(Packet):
-    name = "6LoWPAN Uncompressed IPv6"
+class LoWPAN_HC2_UDP(Packet):
+    name = "6LoWPAN HC1 UDP encoding"
     fields_desc = [
-        BitField("_type", 0x0, 8)
+        BitEnumField("sc", 0, 1, ["In-line", "Compressed"]),
+        BitEnumField("dc", 0, 1, ["In-line", "Compressed"]),
+        BitEnumField("lc", 0, 1, ["In-line", "Compressed"]),
+        BitField("res", 0, 5),
     ]
 
-    def default_payload_class(self, pay):
-        return IPv6
+    def default_payload_class(self, payload):
+        return conf.padding_layer
 
 
-class LoWPANMesh(Packet):
-    name = "6LoWPAN Mesh Packet"
+def _get_hc1_pad(pkt):
+    """
+    Get LoWPAN_HC1 padding
+
+    LoWPAN_HC1 is not recommended for several reasons, one
+    of them being that padding is a mess (not 8-bit regular)
+    We therefore add padding bits that are not in the spec to restore
+    8-bit parity. Wireshark seems to agree
+    """
+    length = 0  # in bits, of the fields that are not //8
+    if not pkt.tc_fl:
+        length += 20
+    if pkt.hc2:
+        if pkt.nh == 1:
+            length += pkt.hc2Field.sc * 4
+            length += pkt.hc2Field.dc * 4
+    return (-length) % 8
+
+
+class LoWPAN_HC1(Packet):
+    name = "LoWPAN_HC1 Compressed IPv6"
     fields_desc = [
-        BitField("reserved", 0x2, 2),
-        BitEnumField("_v", 0x0, 1, [False, True]),
-        BitEnumField("_f", 0x0, 1, [False, True]),
-        BitField("_hopsLeft", 0x0, 4),
-        SixLoWPANAddrField("_sourceAddr", 0x0, length_of=lambda pkt: pkt._v and 2 or 8),  # noqa: E501
-        SixLoWPANAddrField("_destinyAddr", 0x0, length_of=lambda pkt: pkt._f and 2 or 8),  # noqa: E501
+        # https://tools.ietf.org/html/rfc4944#section-10.1
+        ByteField("reserved", 0x42),
+        BitEnumField("sp", 0, 1, ["In-line", "Compressed"]),
+        BitEnumField("si", 0, 1, ["In-line", "Elided"]),
+        BitEnumField("dp", 0, 1, ["In-line", "Compressed"]),
+        BitEnumField("di", 0, 1, ["In-line", "Elided"]),
+        BitEnumField("tc_fl", 0, 1, ["Not compressed", "zero"]),
+        BitEnumField("nh", 0, 2, {0: "not compressed",
+                                  1: "UDP",
+                                  2: "ICMP",
+                                  3: "TCP"}),
+        BitEnumField("hc2", 0, 1, ["No more header compression bits",
+                                   "HC2 Present"]),
+        # https://tools.ietf.org/html/rfc4944#section-10.2
+        ConditionalField(
+            MultipleTypeField(
+                [
+                    (PacketField("hc2Field", LoWPAN_HC2_UDP(),
+                                 LoWPAN_HC2_UDP),
+                        lambda pkt: pkt.nh == 1),
+                    # (PacketField("hc2Field", LoWPAN_HC2_ICMP(),
```

Comment:
Are these comments needed?

Reply:
This PR hasn't implemented ICMP and TCP for the HC1 encoding (it's barely mentionned in the RFC, but also very very rarely used and I just couldn't make it work). I just left the comments of where those features should go

Reply:
Ok. I understand. Could you add a comment, for example `# TODO: ICMP & TCP not implemented yet for the HC! encoding`, before these commented `PacketField`?